### PR TITLE
fix(command-build): allow setting context via CONTEXT env, default to production

### PR DESCRIPTION
**Pull request by erezrokah**
🎉 Thanks for submitting a pull request! 🎉

#### Summary

Fixes https://github.com/netlify/cli/issues/4039

`this.name()` is always defined, so it overrides the environment variable (which is evaluated in `@netlify/config`). Also `@netlify/config` defaults to `production` context, so we should pass `undefined` to allow the default behavior.

---

For us to review and ship your PR efficiently, please perform the following steps:

- [x] Open a [bug/issue](https://github.com/netlify/cli/issues/new/choose) before writing your code 🧑‍💻. This ensures we can discuss the changes and get feedback from everyone that should be involved. If you\`re fixing a typo or something that\`s on fire 🔥 (e.g. incident related), you can skip this step.
- [x] Read the [contribution guidelines](../CONTRIBUTING.md) 📖. This ensures your code follows our style guide and
      passes our tests.
- [x] Update or add tests (if any source code was changed or added) 🧪
- [x] Update or add documentation (if features were changed or added) 📝
- [x] Make sure the status checks below are successful ✅

**A picture of a cute animal (not mandatory, but encouraged)**

### CODE REVIEW

File: src/commands/base-command.js [hunk 1]
```diff
@@ -470,7 +470,10 @@ class BaseCommand extends Command {
       return await resolveConfig({
         config: options.config,
         cwd,
-        context: options.context || this.name(),
+        context:
+          options.context || process.env.CONTEXT || ['dev', 'dev:exec', 'dev:trace'].includes(this.name())
```
Comment (erezrokah) [inline src/commands/base-command.js:474]: This was done in the child dev command classes when we had `oclif`. Not sure if there's a better way to do it now.

File: src/commands/base-command.js [hunk 2]
```diff
@@ -470,7 +470,10 @@ class BaseCommand extends Command {
       return await resolveConfig({
         config: options.config,
         cwd,
-        context: options.context || this.name(),
+        context:
+          options.context ||
+          process.env.CONTEXT ||
+          (['dev', 'dev:exec', 'dev:trace'].includes(this.name()) ? 'dev' : undefined),
```
Comment (erezrokah) [inline src/commands/base-command.js:476]: This was previously done in `oclif` child class, not sure if there's a better way to do it now.

Comment (lukasholzer) [reply src/commands/base-command.js:476]: I don't think either. I would add a small comment here in the code to outline why we are checking for (`(['dev', 'dev:exec', 'dev:trace']` to help our future selfs when touching this code again 😂

**Review by lukasholzer — DISMISSED** [review]
Thanks for adding the tests and having further safety nets :)

File: src/commands/base-command.js (hunk 2 above)
Comment (erezrokah) [reply src/commands/base-command.js:477]: Good point, done in https://github.com/netlify/cli/pull/4040/commits/7ba63676d452b0a5fe8e4a1a6eaa5a2329ebdaf2

**Review by lukasholzer — APPROVED** [review]
Thx for adding the comment! 🎉

**Pull request merged**